### PR TITLE
optional adjustment to dust requirements

### PR DESCRIPTION
**Pull request by lunfardo314**

### CODE REVIEW

File: packages/transaction/makeoutput.go [hunk 1]
```diff
@@ -28,20 +29,23 @@ func ExtendedOutputFromPostData(
...
+// If auto adjustment to dust is disabled and not enough iotas, returns an error
 func MakeExtendedOutput(
 	targetAddress iotago.Address,
 	senderAddress iotago.Address,
 	assets *iscp.Assets,
 	metadata *iscp.RequestMetadata,
 	options *iscp.SendOptions,
 	rentStructure *iotago.RentStructure,
-) *iotago.ExtendedOutput {
+	notAutoAdjustToDustRequirement ...bool,
```
Comment (jorgemmsilva) [inline packages/transaction/makeoutput.go]: nit: feels off to have "negative meaning" boolean values `notAutoAdjustToDustRequirement`

I would normally go for `shouldAutoAdjustDustRequirement` which defaults to true when not specified

**Review by jorgemmsilva — APPROVED** [review]

File: packages/transaction/makeoutput.go (hunk 1 above)
Comment (lunfardo314) [reply packages/transaction/makeoutput.go]: me too. But (a) currently auto adjust is default, so better to stay it this way and (b) `false` is a default value for booleans . So with `false` and absense it should go for default. It means providing `true` we override default and make NOT to autoadjust

Comment (lunfardo314) [reply packages/transaction/makeoutput.go]: I will rename it to `supressAutoAdjustment`

File: packages/transaction/makeoutput.go [hunk 2]
```diff
@@ -28,20 +29,23 @@ func ExtendedOutputFromPostData(
...
+// If auto adjustment to dust is disabled and not enough iotas, returns an error
 func MakeExtendedOutput(
 	targetAddress iotago.Address,
 	senderAddress iotago.Address,
 	assets *iscp.Assets,
 	metadata *iscp.RequestMetadata,
 	options *iscp.SendOptions,
 	rentStructure *iotago.RentStructure,
-) *iotago.ExtendedOutput {
+	suppressAutoAdjustToDustRequirement ...bool,
```
Comment (dessaya) [inline packages/transaction/makeoutput.go]: `disable` instead of `suppress`?

File: packages/transaction/requesttx.go [hunk 3]
```diff
@@ -9,11 +9,12 @@ import (
...
-	UnspentOutputs   []iotago.Output
-	UnspentOutputIDs []*iotago.UTXOInput
-	Requests         []*iscp.RequestParameters
-	RentStructure    *iotago.RentStructure
+	SenderKeyPair                cryptolib.KeyPair
+	UnspentOutputs               []iotago.Output
+	UnspentOutputIDs             []*iotago.UTXOInput
+	Requests                     []*iscp.RequestParameters
+	RentStructure                *iotago.RentStructure
+	SuspendAutoAdjustDustDeposit bool // if true, the minimal dust deposit won't be adjusted automatically
```
Comment (dessaya) [inline packages/transaction/requesttx.go:17]: `Disable` instead of `Suspend`?

**Review by dessaya — APPROVED** [review]

**Pull request merged**